p2p/src/simulated/network.rs

```diff
@@ -300,57 +368,49 @@ impl<E: RNetwork + Spawner + Rng + Clock + Metrics, P: PublicKey> Network<E, P>
                 .get_or_create(&metrics::Message::new(&origin, &recipient, channel))
                 .inc();
 
-            // Sample latency and get current time
-            let latency = Duration::from_millis(link.sampler.sample(&mut self.context) as u64);
-            let now = self.context.current();
-
-            // Get peer bandwidth info and calculate effective bandwidth
-            let (sender_egress_bps, sender_egress_available_at) = {
-                let sender = self.peers.get(&origin).unwrap();
-                (sender.egress_bps, sender.egress_available_at)
-            };
+            // Check bandwidth constraints and determine if the message should be delivered
+            let (sender_has_bandwidth, should_deliver) = {
+                let sender_peer = self.peers.get(&origin).expect("sender must exist");
+                let receiver_peer = self.peers.get(&recipient).expect("receiver must exist");
 
-            let (receiver_ingress_bps, receiver_ingress_available_at) = {
-                let receiver = self.peers.get(&recipient).unwrap();
-                (receiver.ingress_bps, receiver.ingress_available_at)
-            };
+                let sender_has_bandwidth = sender_peer.egress.bandwidth_bps > 0;
```

Comment:
~I wonder how much handling we could remove if we required any bandwidth specification to take a non-zero BPS? That seems like a very reasonable expectation?~

For completeness, I guess it is worth supporting this to ensure a sender burns their egress (may be useful modeling a sybil attack)? If we don't add a link, the message will never be sent to begin with.

That being said, I suppose we could model a sybil as infinite bandwidth rather than 0. So, I guess my point still stands (why do we support 0 bps as a config)?